test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -400,6 +400,80 @@ void ProtoConverter::visit(VarDecl const& _x)
 	}
 }
 
+void ProtoConverter::visit(MultiVarDecl const& _x)
+{
+	m_output << "let ";
+	vector<string> varNames;
+	// We support up to 4 variables in a single
+	// declaration statement.
+	unsigned numVars = _x.num_vars() % 3 + 2;
+	string delimiter = "";
+	for (unsigned i = 0; i < numVars; i++)
+	{
+		string varName = newVarName();
+		varNames.push_back(varName);
+		m_output << delimiter << varName;
+		if (i == 0)
+			delimiter = ", ";
+	}
+	m_output << "\n";
+
+	// If we are inside a for-init block, there are two places
+	// where the visited vardecl may have been defined:
+	// - directly inside the for-init block
+	// - inside a block within the for-init block
+	// In the latter case, we don't scope extend.
+	if (m_inFunctionDef)
+	{
+		// Variables declared directly in for-init block
+		// are tracked separately because their scope
+		// extends beyond the block they are defined in
+		// to the rest of the for-loop statement.
+		if (m_inForInitScope && m_forInitScopeExtEnabled)
+		{
+			yulAssert(
+				!m_funcForLoopInitVars.empty() && !m_funcForLoopInitVars.back().empty(),
+				"Proto fuzzer: Invalid operation"
+			);
+			for (auto const& varName: varNames)
+				m_funcForLoopInitVars.back().back().push_back(varName);
+		}
+		else
+		{
+			yulAssert(
+				!m_funcVars.empty() && !m_funcVars.back().empty(),
+				"Proto fuzzer: Invalid operation"
+			);
+			for (auto const& varName: varNames)
+				m_funcVars.back().back().push_back(varName);
+		}
+
+	}
+	else
+	{
+		if (m_inForInitScope && m_forInitScopeExtEnabled)
```

Comment:
this if can be merged with the else above

Reply:
Thanks. I replaced nested if else with `if...else if ` statements.

Reply:
This clean-up logic also applies to normal variable declaration `let x` and not just multiple variable declarations `let x_0, x_1`, so I applied the same diff to the `VarDecl` visitor as well.

Reply:
Hmm... may be this is a good opportunity to refactor the duplicate code into a utility function that can be called for both single and multiple variable declarations, sorry that I thought about it after requesting review :P

Reply:
Refactoring done.